GarbageCan.Application/XP/Services/XpCalculatorService.cs

```diff
@@ -1,24 +1,37 @@
 ﻿using System;
+using GarbageCan.Application.Boosters.Commands;
 using GarbageCan.Application.Common.Interfaces;
+using MediatR;
 
 namespace GarbageCan.Application.XP.Services
 {
     public class XpCalculatorService : IXpCalculatorService
     {
-        public double XpEarned(string message)
+        private readonly IMediator _mediator;
+
+        public XpCalculatorService(IMediator mediator)
         {
+            _mediator = mediator;
+        }
+
+        public double XpEarned(string message, ulong guildId)
+        {
+            float GetMultiplier() =>
+                _mediator.Send(new GetBoosterMultiplierCommand { GuildId = guildId })
+                    .GetAwaiter()
+                    .GetResult();
```

Comment:
If this is going to do an async operation, it'd be better to expose this as an async method than try to do this synchronously

Reply:
i figured this was probably best or at least not a huge deal because if you look at the command being called its synchronous